### PR TITLE
[CI-6874]: Import TI instrumentation/client code from lite-engine

### PR DESCRIPTION
**Pull request by rutvijmehta-harness**

### CODE REVIEW

File: pipeline/runtime/run.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ package runtime
 import (
 	"context"
 	"fmt"
+	tiCfg "github.com/harness/lite-engine/ti/config"
```
Comment (shubham149) [inline pipeline/runtime/run.go]: Move this to L22

File: pipeline/runtime/runtest.go [hunk 2]
```diff
@@ -7,23 +7,27 @@ package runtime
 import (
 	"context"
 	"fmt"
+	tiCfg "github.com/harness/lite-engine/ti/config"
```
Comment (shubham149) [inline pipeline/runtime/runtest.go]: Ditto

File: pipeline/runtime/step_executor.go [hunk 3]
```diff
@@ -7,6 +7,7 @@ package runtime
 import (
 	"context"
 	"fmt"
+	tiCfg "github.com/harness/lite-engine/ti/config"
```
Comment (shubham149) [inline pipeline/runtime/step_executor.go]: ditto

**Review by shubham149 — APPROVED** [review]

File: pipeline/runtime/run.go (hunk 1 above)
Comment (rutvijmehta-harness) [reply pipeline/runtime/run.go]: Done

File: pipeline/runtime/runtest.go (hunk 2 above)
Comment (rutvijmehta-harness) [reply pipeline/runtime/runtest.go]: Done

File: pipeline/runtime/step_executor.go (hunk 3 above)
Comment (rutvijmehta-harness) [reply pipeline/runtime/step_executor.go]: Done

**Review by shubham149 — APPROVED** [review]

**Pull request merged**